### PR TITLE
Add style guide to front end repo

### PR DESCRIPTION
**Pull request by teallarson**
## What
Moves our front-end style guide from a Google Doc to a Markdown file in our repo.

## How
☝️

## Recommended reading order
1. `x.md`

### CODE REVIEW

**Comment by tealjulia** [comment]
This is unchanged since our last draft in early May.  Is there anything we've learned or agreed upon since then that deserves a spot in here?

File: airbyte-webapp/STYLE-GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Frontend Style Guide
```
Comment (timroes) [inline airbyte-webapp/STYLE-GUIDE.md:1]: Small note, usually I think most repositories call the file `STYLEGUIDE.md` (without the hyphen).

File: airbyte-webapp/STYLE-GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+This serves as a living document regarding conventions we have agreed upon as a frontend team. In general, the aim of these decisions and discussions is to both (a) increase the readability and consistency of our code and (b) decrease day to day decision-making so we can spend more time writing better code.  
+
+## General Code Style and Formatting
+
+* Where possible, we rely on automated systems to maintain consistency in code style 
+* We use eslint, Prettier, and VSCode settings to automate these choices.  The configuration files for these are checked into our repository, so no individual setup should be required beyond ensuring the [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) and [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extensions are installed in VSCode
```
Comment (timroes) [inline airbyte-webapp/STYLE-GUIDE.md:8]: The prettier extension should not be required (I don't have it installed), since we run prettier via ESLint, and it can autofix via it, it should be enough to have the eslint extension installed and the following in your configuration:

```js
"editor.codeActionsOnSave": {
  "source.fixAll.eslint": true,
},
```

File: airbyte-webapp/STYLE-GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+## Exporting
+
+* Export at declaration, not at the bottom. For example:
+    * export const myVar ✅
+    * const myVar; export { myVar }; ❌
+
+
+## File Naming
+
+* TODO
```
Comment (timroes) [inline airbyte-webapp/STYLE-GUIDE.md:23]: Should we remove this bit for now, until we have actual content?

File: airbyte-webapp/STYLE-GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+## Git
+
+* All branches should be &lt;my name>/feature-or-bug-description, for example:
+    * kc/generated-api-client
+    * teal/deleted-connections-ui
+    * edmundo/fix-pwd-input-vis-btn
+
```
Comment (timroes) [inline airbyte-webapp/STYLE-GUIDE.md:32]: Since this is not really related to code-style (and doesn't apply to community contributors), I wonder if we should move this instead separate to the Notion page, maybe under the "Frontend Development Setup" section?

**Review by timroes — APPROVED** [review]
LGTM

**Review by edmundito — APPROVED** [review]
